### DISCUSSION TITLE
Orchestrator Generator Fails with 500

### DISCUSSION THREAD

**Discussion by zachhutchins** [mia-platform/community#537] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.102.8

### Description

When using the Orchestrator Generator I get a 500 response with message

`Cannot read properties of undefined (reading 'error')`

I have confirmed that the underlying service it is calling is returning good payload and status code through logs. The strange part is that this is a flickering error. Is there a timeout that we are hitting for this request and thus causing a failure on your end? I am not able to see any logs that mention failure on your end so it's unclear.

It may be helpful to know the code being logged is `"code":"FAILS_TO_GENERATE_CONFIGURATION_FILES"`, but again that is not consistent with the logs of the service being called.

### Actual Outcome

Failure to deploy when using the advanced workflow + orchestrator generator.

### Expected Outcome

To be able to deploy, or if there is a timeout to be able to configure it.

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @zachhutchins, this is definitely strange. There's a 5s timeout configured, can you confirm you never hit it? If so I can dive a little bit deeper and try to replicate the issue (any additional info for proper replication is welcome).

As for the timeout, at the moment it is not configurable. However, there's room for improvement for sure, if you feel like you need the feature reach out to your Mia-Platform referent :)

**Reply by zachhutchins** [reply]
I can confirm that some of my requests are taking longer than 5s. The main issue is given the nature of the provider it's not unforeseen that that request goes over 5s. Ill reach out to the referent thanks @fredmaggiowski

**Reply by fredmaggiowski** [reply]
HI @zachhutchins, brief update on the matter: by the time we plan and execute a proper feature to make this configurable, we have increased the timeout to 15s, this update will be included in the upcoming v13.1.0 (preview available end of month).

Hope this can help to significantly reduce errors! 

(if you feel like 15s may still be not enough just let me know, we have room for some more seconds 💪)
